src/webgl/3d_primitives.js

```diff
@@ -1024,30 +1024,61 @@ p5.RendererGL.prototype.rect = function(args) {
   return this;
 };
 
-p5.RendererGL.prototype.quad = function(x1, y1, x2, y2, x3, y3, x4, y4) {
+p5.RendererGL.prototype.quad = function(
```

Comment:
generally with functions with variable number of args, we leave the arguments unnamed and test arguments.length. I think this could help condense the code a bit. would you be up for updating this function in this way?

Reply:
I feel like that would be un-doing the work from this commit 11 months ago
https://github.com/processing/p5.js/commit/41025510d7ddeec704613696eaa283291a69e5da
https://github.com/processing/p5.js/commit/41025510d7ddeec704613696eaa283291a69e5da#diff-dfd440848f7f2b0c2e4a2f2a19aa79b1R834

Not sure what the motivation was from changing it from `arguments` to named `args`.

Reply:
@Spongman can you elaborate on the reasoning on this commit?

Reply:
yeah, mostly because it's significantly simpler to read code using named arguments than using the indexed `arguments` array (whose use is generally considered an anti-pattern unless absolutely necessary). most of those methods (while possibly having optional parameters) do not have overloaded meaning for their arguments, so the use of the `arguments` array is not necessary. also because the use of the `arguments` array can also have significant performance implications - often it excludes the function from being JIT compiled and/or inlined.
i think it's important to distinguish between functions that have variable _numbers_ of arguments, and those that have variable _types_ (or meanings) of arguments. the former don't really need to use the indexed `arguments` array. for the latter, it makes more sense.

Reply:
Thanks @Spongman. As a general rule from the beginning, we've actually used named arguments only when there are not variable numbers nor variables types of arguments. That is, there is only one signature. I realize that this isn't entirely consistent throughout, but I'd like to stay with this for now. We are planning an overhaul to ES6 soon, and we can revisit again soon.

Reply:
hmm... i wasn't aware of that. currently, in the p5 library codebase, the use of the `arguments` array is quite definitely the exception rather than the rule. a regex search for ` \* .*@.*.*\[` will show those functions that take optional parameters. very few of them use the `arguments` array.

Reply:
yes, like I said it is not consistent. but I would like to stick to this rule where we can going forward, until we have the ES6 update conversation.

Reply:
Actually, maybe I wasn't clear, or I mixed that up. If the parameters are optional at the end, where the order remains the same, this would be fine to use named ones. But in this case, we're actually talking about a different ordering, as far as I can tell? So we can stick with arguments here. I think we'll actually also want to update this PR with [multiple signatures](https://github.com/processing/p5.js/blob/master/developer_docs/inline_documentation.md#additional-signatures) in the reference too, for the 2D and 3D versions.

Reply:
yup, that's what i was referring to when i said

> i think it's important to distinguish between functions that have variable numbers of arguments, and those that have variable types (or meanings) of arguments

above.

Reply:
I see, got it. I thought you meant different types as in numbers vs strings etc. Cool looks like we're on the same page then. Thanks for clarifying.

Reply:
indeed. sorry, i probably wasn't clear enough before... i'd like to think that ES6 will help with all this, but unfortunately function overloading was left off the table this time around (AS3 didn't support it either).

Reply:
yeah, it won't exactly save us. but I think it will be a good time to document a few of these guidelines together so there's more clarity around code style and patterns.

Reply:
Ok I can't tell where we've landed. I think we've argued for not using `arguments` since this function only has a variable number of args?

> i think it's important to distinguish between functions that have variable numbers of arguments, and those that have variable types (or meanings) of arguments. the former don't really need to use the indexed arguments array.

I think the named arguments would be clearer for now. And any new pattern can be agreed on when we can write in ES6.

Reply:
@unjust we've agreed that named arguments are used in the case where there are either (1) only one signature possible or (2) a variable number of arguments but always the same order and meaning. the arguments object is used in cases where there are variable arguments with different orderings.

so, please remove the named arguments and use the arguments object instead.
also, we will need to add an [additional signature](https://github.com/processing/p5.js/blob/master/developer_docs/inline_documentation.md#additional-signatures) to distinguish the two use cases.

once these two updates have been made, this code can be merged. thanks @unjust!